napari/_qt/containers/_layer_delegate.py

```diff
@@ -120,17 +128,41 @@ def get_layer_icon(
         )  # put icon on the right
         option.features |= option.ViewItemFeature.HasDecoration
 
+    def _paint_loading(
+        self,
+        painter: QPainter,
+        option: QStyleOptionViewItem,
+        index: QtCore.QModelIndex,
+    ):
+        """Paint loading layer indicator."""
+        loaded = index.data(LoadedRole)
+        if not loaded:
+            self._load_movie.start()
+            load_rect = option.rect.translated(4, 8)
+            h = index.data(Qt.ItemDataRole.SizeHintRole).height() - 16
+            load_rect.setWidth(h)
+            load_rect.setHeight(h)
+            painter.drawPixmap(load_rect, self._load_movie.currentPixmap())
+
     def _paint_thumbnail(self, painter, option, index):
         """paint the layer thumbnail."""
         # paint the thumbnail
         # MAGICNUMBER: numbers from the margin applied in the stylesheet to
         # QtLayerTreeView::item
-        thumb_rect = option.rect.translated(-2, 2)
-        h = index.data(Qt.ItemDataRole.SizeHintRole).height() - 4
-        thumb_rect.setWidth(h)
-        thumb_rect.setHeight(h)
-        image = index.data(ThumbnailRole)
-        painter.drawPixmap(thumb_rect, QPixmap.fromImage(image))
+        loaded = index.data(LoadedRole)
+        if loaded:
+            # only pause the loading movie if all the layers are loaded. The
+            # last layer that enters the loaded state will pause the load movie
+            all_loaded = index.model().sourceModel().all_loaded()
+            if all_loaded:
```

Comment:
Does this imply there is only one shared instance of the `QMovie`? If I remember that's because there's only one instance of this delegate shared across all item views, right?

If that's true, I think it's fine, but you might want to add it to the implementation comment above because it explains the underlying reason why we need this specific logic.

Reply:
Totally agree! Will expand the comment :+1: